sdk/core/azure-core/azure/core/pipeline/transport/_base_async.py

```diff
@@ -190,3 +190,16 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         # __exit__ should exist in pair with __enter__ but never executed
         pass  # pragma: no cover
+
+    async def _retrieve_request_data(self, request):
```

Comment:
It's technically specific to our "requests specific async" code, and not generic to the whole world of async like aiohttp.

For the sake of keeping it clean, I would do
```python
# new file _base_requests_async.py
class RequestsAsyncTransportBase(RequestsTransport, AsyncHttpTransport):
    async def _retrieve_request_data(self, request):
        ....

# same file as today
class AsyncioRequestsTransport(RequestsAsyncTransportBase):
    .......

# same file as today
class TrioRequestsTransport(RequestsAsyncTransportBase):
    .....
```

Reply:
Then `AsyncHttpTransport` remains untouched